src/main/java/org/jitsi/videobridge/cc/BitrateController.java

```diff
@@ -1346,10 +1342,14 @@ private TrackBitrateAllocation(
             long idealBps = 0;
             for (RTPEncodingDesc encoding : encodings)
             {
-                if (maxFrameHeight >= 0 && encoding.getHeight() > maxFrameHeight)
+                int idealHeight = videoConstraints.getIdealHeight();
+                if (idealHeight >= 0 && encoding.getHeight() > idealHeight)
                 {
                     continue;
                 }
+
+                boolean selected = videoConstraints
+                    .equals(VideoConstraints.SELECTED_ENDPOINT_CONSTRAINT);
```

Comment:
I thought the `idealHeight` in the constraint obsoletes the idea of explicitly setting something as 'selected'?

Reply:
The `VideoConstraints.SELECTED_ENDPOINT_CONSTRAINT` is meant to be used when translating the selected endpoint messages into video constraints messages for backwards compatibility purposes so that we're not forced to change anything in the client.

Reply:
I misunderstood you, the track allocation should operate on constraints only. Is that your point?

Reply:
I'll mark this conversation as resolved because I think this has now been addressed.